homeassistant/components/bosch_alarm/switch.py

```diff
@@ -0,0 +1,152 @@
+"""Support for Bosch Alarm Panel outputs and doors as switches."""
+
+from __future__ import annotations
+
+from collections.abc import Callable, Coroutine
+from dataclasses import dataclass
+from typing import Any
+
+from bosch_alarm_mode2 import Panel
+from bosch_alarm_mode2.panel import Door
+
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from . import BoschAlarmConfigEntry
+from .const import DOMAIN
+from .entity import BoschAlarmDoorEntity, BoschAlarmOutputEntity
+
+
+@dataclass(kw_only=True, frozen=True)
+class BoschAlarmSwitchEntityDescription(SwitchEntityDescription):
+    """Describes Bosch Alarm door entity."""
+
+    value_fn: Callable[[Door], bool]
+    on_fn: Callable[[Panel, int], Coroutine[Any, Any, None]]
+    off_fn: Callable[[Panel, int], Coroutine[Any, Any, None]]
+
+
+DOOR_SWITCH_TYPES: list[BoschAlarmSwitchEntityDescription] = [
+    BoschAlarmSwitchEntityDescription(
+        key="locked",
+        translation_key="locked",
+        value_fn=lambda door: door.is_locked(),
+        on_fn=lambda panel, door_id: panel.door_relock(door_id),
+        off_fn=lambda panel, door_id: panel.door_unlock(door_id),
+    ),
+    BoschAlarmSwitchEntityDescription(
+        key="secured",
+        translation_key="secured",
+        value_fn=lambda door: door.is_secured(),
+        on_fn=lambda panel, door_id: panel.door_secure(door_id),
+        off_fn=lambda panel, door_id: panel.door_unsecure(door_id),
+    ),
+    BoschAlarmSwitchEntityDescription(
+        key="cycling",
+        translation_key="cycling",
+        value_fn=lambda door: door.is_cycling(),
+        on_fn=lambda panel, door_id: panel.door_cycle(door_id),
+        off_fn=lambda panel, door_id: panel.door_relock(door_id),
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: BoschAlarmConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up switch entities for outputs."""
+
+    panel = config_entry.runtime_data
+    entities: list[SwitchEntity] = [
+        PanelOutputEntity(
+            panel, output_id, config_entry.unique_id or config_entry.entry_id
+        )
+        for output_id in panel.outputs
+    ]
+
+    entities.extend(
+        PanelDoorEntity(
+            panel,
+            door_id,
+            config_entry.unique_id or config_entry.entry_id,
+            entity_description,
+        )
+        for door_id in panel.doors
+        for entity_description in DOOR_SWITCH_TYPES
+    )
+
+    async_add_entities(entities)
+
+
+PARALLEL_UPDATES = 0
```

Comment:
as double check, does the device support multiple commands at once?

Reply:
The python lib handles it, for the panels that work fine with multiple commands it'll send more than one command, and for the ones that don't it'll make sure only one command is sent at a time